src/main/java/org/spongepowered/api/network/ClientPlayerConnection.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.network;
+
+import org.spongepowered.api.entity.living.player.client.ClientPlayer;
+
+/**
+ * Represents a connection of a client to the server where
+ * the {@link ClientPlayer} has successfully joined.
+ */
+public interface ClientPlayerConnection extends PlayerConnection, ClientSideConnection {
+
+    /**
+     * Gets the associated {@link ClientPlayer player} for this connection.
+     *
+     * @return The associated player
+     */
+    @Override
+    ClientPlayer getPlayer();
```

Comment:
Shouldn't this return a `LocalPlayer` since you can't get the connection of a remote player?

Reply:
@Cybermaxke That is a good point, LocalPlayer is "me" in Client engine.

ClientPlayer -> LocalPlayer/RemotePlayer